### PR TITLE
Feature/ec2 instance

### PR DESCRIPTION
**Pull request by nmls**
Re-factored the code:
- all vars are now in one file in vars/ENV/vars.yml;
- var names have been changed;
- yaml code has been reidented;
- playbook.yml renamed to ec2.yml.

### CODE REVIEW

**Comment by wakeful** [comment]
- [ ] remove vars/dev/.vars.yml.swp file

File: vars/dev/vars.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+# aws account details
+aws_region: "{{ lookup('env', 'AWS_REGION') }}"
+
+# ec2 instance default options
+ec2_instance_ami: "ami-e05a4d84"
```
Comment (wakeful) [inline vars/dev/vars.yml]: can we change this AMI to be one of the public [CentOS images](https://wiki.centos.org/Cloud/AWS)?

File: vars/dev/vars.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ec2_instance_pub_ip: "yes"
+ec2_instance_volume_size: 50
+ec2_instance_tag_name: "YAAPi ec2"
+ec2_instance_keypair: "nicholas"
```
Comment (wakeful) [inline vars/dev/vars.yml]: probably worth mentioning that this key-pair needs to be already added in AWS

File: roles/ec2-create/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    vpc_subnet_id: "{{ created_pub_subnet_id }}"
+    group_id: "{{ created_secgrp_id }}"
+    assign_public_ip: "{{ ec2_instance_pub_ip }}"
+    volumes:
+      - device_name: /dev/sda1
+        volume_size: "{{ ec2_instance_volume_size }}"
+        delete_on_termination: true
+    count: 1
+    wait: yes
+    instance_tags:
```
Comment (wakeful) [inline roles/ec2-create/tasks/main.yml]: can we also add **env** tag?

File: roles/ec2-create/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    keypair: "{{ ec2_instance_keypair }}"
+    instance_type: "{{ ec2_instance_type }}"
+    image: "{{ ec2_instance_ami }}"
+    vpc_subnet_id: "{{ created_pub_subnet_id }}"
+    group_id: "{{ created_secgrp_id }}"
+    assign_public_ip: "{{ ec2_instance_pub_ip }}"
+    volumes:
+      - device_name: /dev/sda1
+        volume_size: "{{ ec2_instance_volume_size }}"
+        delete_on_termination: true
```
Comment (wakeful) [inline roles/ec2-create/tasks/main.yml]: please extract **delete_on_termination** into variable

File: roles/ec2-create/tasks/main.yml (hunk 3 above)
Comment (nmls) [reply roles/ec2-create/tasks/main.yml]: Sure. We should be adding some tags (env, project, ...) to all our AWS resources (where possible) in general.

File: vars/dev/vars.yml (hunk 2 above)
Comment (nmls) [reply vars/dev/vars.yml]: If we create a VPC (YAPPi vpc) which is used by the EC2 instance, then we can create a keypair (YAPPi keypair) aswell.

**Comment by nmls** [comment]
Key-pair name for EC2 instance is now taken from 'AWS_KEYPAIR' env variable (.env.example file has been updated). For now, it's all right to have one key-pair for all EC2 instances (besides, var can be always overwritten).
Also, 'delete_on_termination' is now parametrized and defined in vars/dev/vars.yml.

**Comment by nmls** [comment]
Base AMI for EC2 instance is now taken from a CSV file containing all CentOS 7 AMIs. I haven't tested. Can someone do that and eventually fix it? Thanks.

**Comment by wakeful** [comment]
rebase and conflict resolution.

**Pull request merged**